pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractLombokAwareRule.java

```diff
@@ -81,35 +83,14 @@ protected boolean hasClassLombokAnnotation() {
 
     /**
      * Checks whether the given node is annotated with any lombok annotation.
-     * The node can be any node, e.g. class declaration or field declaration.
-     * 
+     * The node should be annotateable.
+     *
      * @param node
-     *            the node to check
+     *            the Annotateable node to check
      * @return <code>true</code> if a lombok annotation has been found
      */
-    protected boolean hasLombokAnnotation(Node node) {
-        boolean result = false;
-        Node parent = node.jjtGetParent();
-        List<ASTAnnotation> annotations = parent.findChildrenOfType(ASTAnnotation.class);
-        for (ASTAnnotation annotation : annotations) {
-            ASTName name = annotation.getFirstDescendantOfType(ASTName.class);
-            if (name != null) {
-                String annotationName = name.getImage();
-                if (lombokImported) {
-                    if (LOMBOK_ANNOTATIONS.contains(annotationName)) {
-                        result = true;
-                    }
-                } else {
-                    if (annotationName.startsWith(LOMBOK_PACKAGE + ".")) {
-                        String shortName = annotationName.substring(LOMBOK_PACKAGE.length() + 1);
-                        if (LOMBOK_ANNOTATIONS.contains(shortName)) {
-                            result = true;
-                        }
-                    }
-                }
-            }
-        }
-        return result;
+    protected boolean hasLombokAnnotation(Annotateable node) {
+        return node.isAnyAnnotationPresent(LOMBOK_ANNOTATIONS);
     }
 
     protected ASTAnnotation getLombokAnnotation(Node node, String lombokAnnotation) {
```

Comment:
this method should also benefit from the new logic, at least checking if the node is `Annotateable` to rely tree transversal to the node

Reply:
Done.